plugins/@grouparoo/intercom/src/lib/export-contacts/exportProfile.ts

```diff
@@ -386,20 +395,24 @@ function addCreationPayload(
 
 function addUpdatePayload(
   payload: any,
+  syncOperations: DestinationSyncOperations,
   destinationOptions: SimpleDestinationOptions
 ) {
+  if (!syncOperations.update) {
+    throw new InfoError("Destination is not updating contacts");
+  }
+
   const { creationMode } = destinationOptions;
   const external_id = payload.external_id || null;
   const extra: any = {};
   switch (creationMode) {
     case CreationMode.User:
     case CreationMode.Lead:
-    case CreationMode.Enrich:
       // no changes
       break;
     case CreationMode.Lifecycle:
       // make sure it's the right one
-      if (external_id) {
+      if (external_id && syncOperations.create) {
```

Comment:
When `creationMode` was set to `Enrich`, this would not run. Why is this the case? Does setting the role to user with an external id create a new contact? There's also a test that intentionally checks for this behavior:
https://github.com/grouparoo/grouparoo/blob/51c7f9e431a5d00e934aed9aa811cb7b4b593cb5/plugins/%40grouparoo/intercom/__tests__/integration/export-profile-no-create.ts#L340

@bleonard do you remember why this is the case?